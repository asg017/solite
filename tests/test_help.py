def test_help(solite_cli, snapshot):
    assert solite_cli(["--help"]).stdout == snapshot(name="--help")
    assert solite_cli(["run", "--help"]).stdout == snapshot(name="run --help")
    assert solite_cli(["query", "--help"]).stdout == snapshot(name="query --help")
    assert solite_cli(["repl", "--help"]).stdout == snapshot(name="repl --help")
    assert solite_cli(["jupyter", "--help"]).stdout == snapshot(name="jupyter --help")
