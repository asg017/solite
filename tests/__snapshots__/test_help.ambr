# serializer version: 1
# name: test_help[--help]
  '''
  Solite CLI
  
  Usage: solite [COMMAND]
  
  Commands:
    run      Run SQL scripts
    repl     Start a REPL on a SQLite database
    query    Run a read-only SQL query and output results to a file
    execute  Execute a write SQL statement on a database
    snap     Snapshot testing for extensions and SQL statements
    jupyter  Manage the Solite Jupyter kernel
    docs     Tooling for documenting SQLite extensions
    bench    Run benchmarks on SQL statements
    mcp      MCP
    codegen  Codegen SQL queries into an intermediate representation
    help     Print this message or the help of the given subcommand(s)
  
  Options:
    -h, --help     Print help
    -V, --version  Print version
  
  '''
# ---
# name: test_help[jupyter --help]
  '''
  Manage the Solite Jupyter kernel
  
  Usage: solite jupyter <COMMAND>
  
  Commands:
    install  
    up       
    help     Print this message or the help of the given subcommand(s)
  
  Options:
    -h, --help  Print help
  
  '''
# ---
# name: test_help[query --help]
  '''
  Run a read-only SQL query and output results to a file
  
  Usage: solite query [OPTIONS] <STATEMENT> [DATABASE]
  
  Arguments:
    <STATEMENT>  
    [DATABASE]   
  
  Options:
    -o, --output <OUTPUT>
            
    -f, --format <FORMAT>
            [possible values: csv, tsv, json, ndjson, value, clipboard]
    -p, --parameters <PARAMETERS> <PARAMETERS>
            
    -h, --help
            Print help
  
  '''
# ---
# name: test_help[repl --help]
  '''
  Start a REPL on a SQLite database
  
  Usage: solite repl [DATABASE]
  
  Arguments:
    [DATABASE]  
  
  Options:
    -h, --help  Print help
  
  '''
# ---
# name: test_help[run --help]
  '''
  Run SQL scripts
  
  Usage: solite run [OPTIONS] [DATABASE] [SCRIPT]
  
  Arguments:
    [DATABASE]  
    [SCRIPT]    
  
  Options:
    -p, --parameters <PARAMETERS> <PARAMETERS>  
        --trace <TRACE>                         
    -h, --help                                  Print help
  
  '''
# ---
